src/iconv.jl

```diff
@@ -1,20 +1,76 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, GeneralError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+const UNKNOWN_ERR      = "<<1>> (<<2>>)"
+const OUTPUT_BUFFER    = "ran out of space in the output buffer"
+const UNSUPPORTED_CONV =
+    "conversion from <<1>> to <<2>> not supported by iconv implementation, check that specified encodings are correct"
+const INVALID_SEQUENCE    = "byte sequence 0x<<1>> is invalid in source encoding or cannot be represented in target encoding"
+const INCOMPLETE_SEQUENCE = "incomplete byte sequence at end of input"
+const INVALID_BYTES       = "invalid byte sequence in input"
+
+abstract StringEncodingError
+
+# General purpose error handling type, which supports variable number of arguments,
+# and can be extended to work with different locales
+
+type GeneralError <: Exception
```

Comment:
Right, but this does nothing more than what `UnicodeError` already does in Base, to allow for the format strings to be constant - which is what allows the error handling to be done without forcing the entire function to always push a GC frame (I know that Yuyichao has talked about changing things so that that is not so bad, by only doing that along branches that do the creating, but I'm not sure what the status of the in v0.4.x or master).
Yes, the whole locale business is something to think about for the future, but I think having `GeneralError` which greatly simplifies making the specific exception types, is well worth having here.